db/crud.go

```diff
@@ -1079,11 +1079,14 @@ func (db *DatabaseCollectionWithUser) PutExistingCurrentVersion(ctx context.Cont
 		}
 		matchRev = doc.CurrentRev
 	}
-	generation, _ := ParseRevID(ctx, matchRev)
-	if generation < 0 {
+
+	// set up revTreeID for backward compatibility
+	previousRevTreeID := doc.CurrentRev
+	newGeneration, _ := ParseRevID(ctx, matchRev)
+	if newGeneration < 0 {
 		return nil, nil, "", base.HTTPErrorf(http.StatusBadRequest, "Invalid revision ID")
 	}
-	generation++
+	newGeneration++
```

Comment:
I thought we needed to (also?) do this inside the loop to account for CAS retry.